action_plugins/play_sound/__init__.py

```diff
@@ -141,28 +141,36 @@ def __init__(
         super().__init__(behavior_type)
 
         # Model variables
-        self.sound_filename: str = ""
-        self.sound_volume: int = 50
+        self.soundFilename: str = ""
+        self.soundVolume: int = 50
 
     def _from_xml(self, node: ElementTree.Element, library: Library) -> None:
         self._id = util.read_action_id(node)
-        self.sound_filename = util.read_property(
-            node, "play-sound", PropertyType.String
+
+        self.soundFilename = util.read_property(
+            node, "filename", PropertyType.String
+        )
+        self.soundVolume = util.read_property(
+            node, "volume", PropertyType.Int
         )
 
         if not self.is_valid():
-            raise GremlinError(f"{self.sound_filename} does not exists or is not accessible.")
+            raise GremlinError(f"{self.soundFilename} does not exists or is not accessible.")
 
     def _to_xml(self) -> ElementTree.Element:
         node = util.create_action_node(PlaySoundData.tag, self._id)
-        node.append(util.create_property_node(
-            "play-sound", self.sound_filename, PropertyType.String
-        ))
+        util.append_property_nodes(
+            node,
+            [
+                ["filename", self.soundFilename, PropertyType.String],
+                ["volume", self.soundVolume, PropertyType.Int],
+            ]
+        )
         return node
 
     def is_valid(self) -> bool:
-        if len(self.sound_filename) > 0 and os.path.isfile(self.sound_filename) and \
-           os.access(self.sound_filename, os.R_OK):
+        if len(self.soundFilename) > 0 and os.path.isfile(self.soundFilename) and \
```

Comment:
If you think this "does that file exist and can I access it" might be useful more generally, feel free to extract it out into a function in `gremlin.utils` as you've also needed this functionality in the load plugin action.

Reply:
Good idea. Since I've used the same piece in Load Profile I'd prefer doing it as a separate PR. OK?